libtrip-notification/src/main/java/com/mapbox/navigation/trip/notification/MapboxTripNotification.kt

```diff
@@ -9,33 +9,50 @@ import android.content.Context
 import android.content.Intent
 import android.content.IntentFilter
 import android.os.Build
+import android.text.SpannableString
+import android.text.TextUtils
+import android.text.format.DateFormat
 import android.widget.RemoteViews
 import androidx.core.app.NotificationCompat
 import androidx.core.content.ContextCompat
 import com.mapbox.annotation.navigation.module.MapboxNavigationModule
 import com.mapbox.annotation.navigation.module.MapboxNavigationModuleType
+import com.mapbox.navigation.base.banner.BannerInstruction
+import com.mapbox.navigation.base.options.NavigationOptions
+import com.mapbox.navigation.base.route.model.LegStepNavigation
+import com.mapbox.navigation.base.route.model.RouteProgressNavigation
 import com.mapbox.navigation.base.trip.TripNotification
-import com.mapbox.navigation.base.trip.model.RouteProgress
+import com.mapbox.navigation.trip.notification.utils.distance.DistanceFormatter
+import com.mapbox.navigation.trip.notification.utils.time.TimeFormatter.formatTime
 import com.mapbox.navigation.utils.END_NAVIGATION_ACTION
 import com.mapbox.navigation.utils.NAVIGATION_NOTIFICATION_CHANNEL
 import com.mapbox.navigation.utils.NOTIFICATION_CHANNEL
 import com.mapbox.navigation.utils.NOTIFICATION_ID
 import com.mapbox.navigation.utils.SET_BACKGROUND_COLOR
+import com.mapbox.navigation.utils.extensions.ifNonNull
+import java.util.Calendar
 
 @MapboxNavigationModule(MapboxNavigationModuleType.TripNotification, skipConfiguration = true)
 class MapboxTripNotification(
-    private val applicationContext: Context
+    private val applicationContext: Context,
+    private val distanceFormatter: DistanceFormatter,
```

Comment:
Should we leave the primary constructor as it was (accepting only the context) and add these new parameters as part of the overloaded constructor (so we don't leak these out of the `libtrip-notification` module)?

cc @LukasPaczos

Reply:
In order to show the `arrival distance` with proper `decimal rounding` and `metric unit`, we need to have the `DistanceFormatter` or some sort of `Formatter` that can take a raw distance(`Double`) and convert it to `String` with proper value in order to be displayed in the `TextView` in `notification`. 

There are 2 ways of creating this `DistanceFormatter`
- Inject it in the constructor
- instantiate it inside `MapboxTripNotification/init` method.

If we look at the legacy code, an object of `DistanceFormatter` is being created lot of places including the `UI` library. The way I envisioned it is, after we release 1.0, and start to work on ramping up the UI library, it will use `MapboxTripSession`, which in turn will create the `MapboxTripService` and hence the `MapboxTripNotification`. We can create one instance of `DistanceFormatter` in the `MapboxTripSession` and pass it to the constructor of `MapboxTripSession`. Hence the object of `DistanceFormatter` will be created with creating of TripSession(i.e. when navigation starts) and will destroy with termination of TripSession(i.e. when navigation ends). 

I don't like creating an instance of `DistanceFormatter` inside `init` in case we overload the constructor, because then we may have multiple different instances of `DistanceFormatter` floating everywhere. With adding it to the primary constructor, and injecting it from `MapboxTripSession` we can control it better to have just one instance and add a new method to interface `TripSessiion` -> `fun getDistanceFormatter: DistanceFormatter`

Let me know your thoughts on this

Reply:
Can't `DistanceFormatter` be an `object`?

Reply:
If you look at the functions in `DistanceFormatter`, you need to know the following to be able to convert `Distance(Double)` to `String`
- applicationContext,
- language,
- unitType,
- roundingIncrement

If we convert this to `Object`, how do we pass the `applicationContext`? We can use a setter, but that's not ideal. Same question goes for other 3 parameters

Reply:
The formatter is local to the `libtrip-notification` so we cannot expect it to be passed from the core, because the whole module can be excluded.